### PR TITLE
feat: update cql observation handling

### PR DESCRIPTION
**Pull request by smly**
Resolves https://github.com/smly/RiichiEnv/issues/80

I have implemented the `mask()` method in the `Observation` class in `native/src/observation.rs`.
The new method returns a byte array (`uint8`) of length 82, where indices corresponding to `legal_actions` are set to 1. This allows for efficient masking in Python/PyTorch without manual iteration.

### CODE REVIEW

File: demos/ml_baseline/train_cql.py [hunk 1]
```diff
@@ -29,13 +29,10 @@ def cql_loss(q_values, current_actions, masks=None):
...
-        invalid_mask = (masks < 0.5)
-        q_masked = q_values.clone()
-        q_masked = q_masked.masked_fill(invalid_mask, -1e9)
-        logsumexp_q = torch.logsumexp(q_masked, dim=1)
-    else:
-        logsumexp_q = torch.logsumexp(q_values, dim=1)
+    invalid_mask = (masks == 0)
+    q_masked = q_values.clone()
+    q_masked = q_masked.masked_fill(invalid_mask, -1e9)
+    logsumexp_q = torch.logsumexp(q_masked, dim=1)
```
[the remark this answers is not in the document]
Comment (smly) [reply demos/ml_baseline/train_cql.py]: Fixed in https://github.com/smly/RiichiEnv/pull/81/commits/cd92509a93e55efb0eaf73465b965fc491773511
